library/kotlin/src/io/envoyproxy/envoymobile/Client.kt

```diff
@@ -0,0 +1,73 @@
+package io.envoyproxy.envoymobile
+
+import java.nio.ByteBuffer
+
+interface Client {
+  fun startStream(request: Request, streamCallback: StreamCallback): StreamEmitter
+}
+
+interface StreamCallback {
+  fun onHeaders(statusCode: Int, headers: Map<String, List<String>>)
+  fun onData(byteBuffer: ByteBuffer, endStream: Boolean)
+  fun onMetadata(metadata: Map<String, List<String>>, endStream: Boolean)
```

Comment:
Guessing we probably don't need this in the initial implementation 🤔

Reply:
We probably don't but I kept it here since I was planning to close this in favor of a different PR for submitting

My original intent was to show case what we would have and what is most consistent with the common layer

Reply:
Per offline convo, we should keep this